packages/aws-cdk-lib/aws-ec2/test/vpc.test.ts

```diff
@@ -708,25 +708,73 @@ describe('vpc', () => {
       }).toThrow(/Vpc supports 'availabilityZones' or 'maxAzs', but not both./);
     });
 
-    test('with availabilityZones set correctly', () => {
```

Comment:
Changing existing tests is usually a sign of breaking changes...can you explain why its needed?

Reply:
The reason for any changes in the unit tests were due to tests being run against 'dummy' values. From what I can tell, the only time we can effectively evaluate whether the provided availability zones are valid for the stack are when the context has the availability zones. Otherwise, we're checking against tokens or 'dummy' values. Given your previous comment I reverted the unit tests that I changed.